src/js/components/ChromecastButton.js

```diff
@@ -52,7 +52,7 @@ ChromecastButton = {
     * @see {@link http://docs.videojs.com/Button.html#buildCSSClass|Button#buildCSSClass}
     */
    buildCSSClass: function() {
-      return 'vjs-chromecast-button ' + (this._isChromecastConnected ? 'vjs-chromecast-casting-state ' : '') +
+      return 'vjs-chromecast-button ' + ((cast.framework.CastContext.getInstance().getCastState() === 'CONNECTED') ? 'vjs-chromecast-casting-state ' : '') +
```

Comment:
1. Is this property used anywhere else?
2. Why the property isn't updated?
3. This function might be called quite frequently. Is it okay to poll chromecast API often?

Reply:
1. this._isChromecastConnected is only used for this check
2. I think it's a race condition, subscription in plugin happens after videojs emits `chromecastConnected` event
3. This function is only called on construction and if `chromecastDisconnected` event is propageted so i think it's fine.

Reply:
1. Then delete all references of this property?

Reply:
https://github.com/standardbroadcast/videojs-chromecast/pull/12/commits/c5bca0c290dd40c90a5c9777e5fe3f75111d4869

Reply:
I still have a funny feeling about this change

Could you double checke that button state stays in sync and that this function is rarely called in bunch of cases (page reload, starting casting before playback, starting casting after playback, ending casting after every previous case)?

Reply:
I found that if you refresh, even tough `onChromecastConnected` event was triggered `cast.framework.CastContext.getInstance().getCastState()` was returning `NOT_CONNECTED` so I've added a flag. 
In all the scenarios it seems to stay in sync, the function is only called when events are triggered that means starting/stopping casting or on render, so it's not often.

https://github.com/standardbroadcast/videojs-chromecast/pull/12/commits/05447748ddf6daf2bdee892b768d71b647e31760

Reply:
Actually we can check the cast state only if it wasn't triggered by an event
https://github.com/standardbroadcast/videojs-chromecast/pull/12/commits/fe7329c90b60fe4180a3e65367627f4644b05236